batch2/batch/front_end/front_end.py

```diff
@@ -283,7 +285,22 @@
                 if 'memory' not in resources:
                     resources['memory'] = BATCH_JOB_DEFAULT_MEMORY
 
-                cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_memory_bytes = parse_memory_in_bytes(resources['memory'])
+
+                if req_cores_mcpu == 0:
+                    raise web.HTTPBadRequest(
+                        reason=f'bad resource request for job {id}: '
+                        f'cpu cannot be 0')
+
+                cores_mcpu = adjust_cores_for_memory_request(req_cores_mcpu, req_memory_bytes, WORKER_TYPE)
+
+                if cores_mcpu > WORKER_CORES * 1000:
+                    total_memory_available = round(worker_memory_per_core_gb(WORKER_TYPE) * WORKER_CORES, 2)
```

Comment:
I don't think we should round.  The memory per core in GB is generally a nice, round number, and if you round, you're going to be able to generate messages like: unsatisfiable: requested: 3.75G, maximum: 3.75GB (where the maximum is e.g. 3.749G).